sunpy/timeseries/timeseriesbase.py

```diff
@@ -544,6 +550,15 @@ def truncate(self, a, b=None, int=None):
             start = a
             end = b
 
+        min_time, max_time = self._data.index.min(), self._data.index.max()
+        # Check if the timerange overlaps with the data timerange
+        if not (isinstance(start, builtins.int) and isinstance(end, builtins.int)):
```

Comment:
now it will also with
```
a = pd.Timestamp('2011-06-07 06:55:59')
b = pd.Timestamp('2011-06-07 08:55:59')
```

for out of bounds/partial timeranges